api/envoy/api/v2/core/health_check.proto

```diff
@@ -267,4 +267,10 @@ message HealthCheck {
   // initial health check failure event will be logged.
   // The default value is false.
   bool always_log_health_check_failures = 19;
+
+  // Specifies the ALPN protocol for health check connections. This is useful if the corresponding
+  // upstream is using ALPN-based :ref:`FilterChainMatch <envoy_api_msg_listener.FilterChainMatch>`
+  // along with different protocols for health checks versus data connections.
+  // If empty, no ALPN protocol will be set on HC connections.
+  string alpn_protocol = 21;
```

Comment:
My only concern with this is that this field is specific to TLS. Should we be thinking about some type of generic TLS override structure to make this more future proof? cc @htuch @lizan for thoughts on this.

Reply:
That sounds like a good idea too, I'll update it (while we wait for more feedback on this).

Reply:
added `TLSOptions` (in the absence of a better name), to encapsulate this.